src/app/screens/home/index.tsx

```diff
@@ -163,13 +163,43 @@ function Home() {
   );
   const [choseToVerifyAddresses, setChoseToVerifyAddresses] = useState(false);
   const { isLoading: loadingStxWalletData, isRefetching: refetchingStxWalletData } =
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    stxAddress ? useStxWalletData() : { isLoading: false, isRefetching: false };
+    useStxWalletData();
   const { isLoading: loadingBtcWalletData, isRefetching: refetchingBtcWalletData } =
     useBtcWalletData();
   const { isLoading: loadingCoinData, isRefetching: refetchingCoinData } = useCoinsData();
   const { isLoading: loadingBtcCoinData, isRefetching: refetchingBtcCoinData } =
     useBtcCoinBalance();
+
+  // these will be used to show loaders on the screen
+  // they initialise to true if the data is being loaded or refetched on first render
+  const [displayLoader, setDisplayLoader] = useState({
+    stx: loadingStxWalletData || refetchingStxWalletData,
+    btc: loadingBtcWalletData || refetchingBtcWalletData,
+    coin: loadingCoinData || refetchingCoinData,
+    btcCoin: loadingBtcCoinData || refetchingBtcCoinData,
+  });
+
+  // we update the above loaders here
+  // if a loader is set to false, that means that we have data and we don't want to show the loader on a refetch
+  // so we tie false values to false
+  useMemo(() => {
+    setDisplayLoader(({ stx, btc, coin, btcCoin }) => ({
+      stx: !stx ? false : loadingStxWalletData || refetchingStxWalletData,
+      btc: !btc ? false : loadingBtcWalletData || refetchingBtcWalletData,
+      coin: !coin ? false : loadingCoinData || refetchingCoinData,
+      btcCoin: !btcCoin ? false : loadingBtcCoinData || refetchingBtcCoinData,
+    }));
+  }, [
+    loadingStxWalletData,
+    refetchingStxWalletData,
+    loadingBtcWalletData,
+    refetchingBtcWalletData,
+    loadingCoinData,
+    refetchingCoinData,
+    loadingBtcCoinData,
+    refetchingBtcCoinData,
+  ]);
+
```

Comment:
what if we just stop displaying loading states on isRefetching? that would stop the annoying loading states on refocus.

I'm also not sure we need to trigger refetches on every initial render of home dashboard, but if we do, we could do something like query invalidations instead?

not sure if you've seen these already, but react-query has utils like: 

refetchOnWindowFocus: https://tanstack.com/query/v4/docs/react/guides/window-focus-refetching
query invalidation: https://tanstack.com/query/v4/docs/react/guides/query-invalidation
mutation invalidation: https://tanstack.com/query/v4/docs/react/guides/invalidations-from-mutations

Reply:
I initially stopped displaying on refetch, but if the query lib has a value in cache, the loading status never seems to be true again. It always refetches. This means that if I close the popup and then open it again in a few minutes/hours/days during which time I've received funds, the popup initially shows the old balance and then updates to the correct one once the refetch is done.

I didn't consider the TTL of the cache though. We could lower it and I think it will use the loading state in that case. Will play around a bit 👍